app/jobs/ingest_job.rb

```diff
@@ -0,0 +1,17 @@
+class IngestJob < Hyrax::ApplicationJob
+  queue_as Hyrax.config.ingest_queue_name
+
+  after_perform do |job|
+    opts = job.arguments.reverse.detect { |x| x.is_a? Hash } || {}
+    wrapper = job.arguments.first
+    ContentNewVersionEventJob.perform_later(wrapper.file_set, wrapper.user) if opts[:notification]
```

Comment:
Correct, event jobs do not take the same args or the same number of args.  That latter idea sounds like making the thing you're objecting to worse, no?  Also the caller doesn't get to decide what *type* of action this job considers itself to implement.  If we wanted to go that way (I don't), we would cut out the boolean, and just use the presence of a `notification_class` to signal `true`.